src/hotspot/share/gc/shared/space.inline.hpp

```diff
@@ -123,6 +125,7 @@ class DeadSpacer : StackObj {
       log_develop_trace(gc, compaction)("Inserting object to dead space: " PTR_FORMAT ", " PTR_FORMAT ", " SIZE_FORMAT "b",
           p2i(dead_start), p2i(dead_end), dead_length * HeapWordSize);
 
+      _dead_space += dead_length;
```

Comment:
I do not think adding this to the counter here instead of the other method for every object makes a difference performance-wise.

As mentioned before, `_allowed_deadspace_words` counts *down* from `(space->capacity() * ratio / 100) / HeapWordSize;` to whatever end value.

So at the end of collection, `(space->capacity() * ratio / 100) / HeapWordSize - _allowed_deadspace_words` should be equal to what `_dead_space` is now.

Please add a getter to `DeadSpacer` that calculates this (factoring out the calculation of the maximum allowed deadspace).

Reply:
Fixed in https://github.com/openjdk/jdk/pull/2579/commits/f708023b84e33aee34f92d183a0d03d2747646db